### PR TITLE
Upgrade to Elasticsearch 6

### PR DESCRIPTION
**Pull request by markbreedlove**
Remove Tire gem and modify code that generates queries and formats results to work with Elasticsearch 6.

Prune obsolete code that is either unnecessary with the removal of Tire, or that is outdated for other reasons.

Wherever possible, API behavior, including the formatting of responses, has been unchanged. There are some subtle differences, though, in the formatting and ordering of date facets. Elasticsearch 6 has a significantly different facets API than version 0.90; in fact, they're not even called facets anymore, but "aggregations."  I've kept the term "facets" in the DPLA API. Facet results are formatted differently in Elasticsearch 6 and I've attempted to massage the results to be as similar as possible to earlier API results. Ordering of facets by date was inconsistent before now; if you used the 'day' or 'year' date modifier (which isn't documented in the API Codex), ordering would be by count per bucket, but if you used 'decade' or 'century' it would be by date. I believe the latter is the correct way for a so-called histogram facet, so now all of the date facets are ordered by date.

*Pull Request Reviewers*: You're probably best off starting with the unit tests in `v1/spec`. This is the best documentation of how the API works, and the best documentation of what was changed. I'll pass along information on where you can find this staged on a server.

### CODE REVIEW

**Comment by markbreedlove** [comment]
I've discovered a bug in faceting on `admin.contributingInstitution`, which I am working on fixing ...

**Comment by markbreedlove** [comment]
OK, the bugs I've known of up until this point are fixed.

**Comment by mgiraldo** [comment]
i suppose the lack of facets in the [empty search results](https://demo.dp.la/search?q=) is due to an incomplete index and not some other reason? the code postlight did shows a facet only if there are results to show if selected

**Comment by AudreyAltman** [comment]
I see your comment that a plan is in place to create a new mapping on the indexing end for what was the `admin.contributingInstitution` compound field.  I think you know this already, but just in case I'll mention that the `dpla-frontend` app makes active use of `admin.contributingInstitution` so we need to be careful about timing that change.

File: Gemfile.lock [hunk 1]
```diff
@@ -7,7 +7,6 @@ PATH
       httparty
       json
       rails
-      tire (= 0.6.2)
```
Comment (AudreyAltman) [inline Gemfile.lock]: 🎊

File: v1/spec/lib/v1/field_spec.rb [hunk 2]
```diff
@@ -6,43 +6,41 @@ module V1
     let(:resource) { 'item' }
     let(:item_mapping) {
       {
-        'id' => { 'type' => 'string', 'index' => 'not_analyzed', 'sort' => 'field' },
-        'title' => { 'type' => 'string' },
-        'description' => { 'type' => 'string' },
+        'id' => { 'type' => 'keyword', 'sort' => 'field' },
+        'title' => { 'type' => 'text' },
+        'description' => { 'type' => 'text' },
```
Comment (AudreyAltman) [inline v1/spec/lib/v1/field_spec.rb]: These new `text` and `keyword` types are nice - looks like the schema will be easier to maintain now.

**Comment by markbreedlove** [comment]
Hi, @AudreyAltman. Regarding `admin.contributingInstitution`, made the changes after testing it out with the new frontend and having it not work, with errors related to that field. The field still exists in the Elasticsearch index, in that you can search and facet on it, but it's not part of the `_source`, i.e. the stored document.

**Comment by mgiraldo** [comment]
see also dpla/dpla-frontend#510

**Pull request merged**